### ISSUE TITLE
Consistency of dictionary iteration behavior between Python 2 and 3

### ISSUE THREAD

**Issue by Kenadia** [google/openhtf#688]
#679 introduced support for Python 3, in many cases converting dict iteration code, e.g. `d.itervalues()` -> `d.values()`. In many cases, we are left with code that has different semantics in Python 2 and 3, e.g. `d.values()` produces a list in Python 2, and a view object in Python 2.

This doesn't currently appear to be a problem, but down the line, so long as we maintain OpenHTF as a hybrid Python 2/3 project, this may make things harder for maintainers, who will have to consider different behaviors between the two versions.

[PEP 469](https://www.python.org/dev/peps/pep-0469/) discusses this problem and proposes solutions. For example, helper functions can be used to succinctly convert Python 2 code to hybrid code:
```
d.keys() -> list(d)
d.values() -> listvalues(d)
d.items() -> listitems(d)
d.iterkeys() -> iter(d)
d.itervalues() -> itervalues(d)
d.iteritems() -> iteritems(d)
```

**Comment by fahhem** [comment]
six is a single file library that provides these helpers and many others.
If we don't want to add it as a dependency, then we can include it inside
openhtf. As long as we update for security patches then it's safe/okay.

**Comment by Kenadia** [comment]
Partially addressed by #737
